include/fmt/compile.h

```diff
@@ -515,7 +515,7 @@ constexpr parse_specs_result<T, Char> parse_specs(basic_string_view<Char> str,
   auto ctx = basic_format_parse_context<Char>(str, {}, arg_id + 1);
   auto f = formatter<T, Char>();
   auto end = f.parse(ctx);
-  return {f, pos + (end - str.data()) + 1, ctx.next_arg_id()};
+  return {f, pos + static_cast<decltype(pos)>(end - str.data()) + 1, ctx.next_arg_id()};
```

Comment:
@vitaut I assume this isn't the solution that you want here - I just went with this to have something to start the discussion with. Ultimately, the problem here is that `end` and `str.data()` are both pointer types and the result of subtraction of pointer types is `ptrdiff_t`, which is `signed long` on x86_64. The addition to `pos`, which is of an unsigned type, causes a sign conversion error.

I'm fairly confident that the `ptrdiff_t` would never be a negative type, but there's nothing binding that behavior.

Hopefully that explanation makes sense - in light of that how would you like to proceed here?

Reply:
Here's a minimal repro of the warning if you'd like to play around with it. https://godbolt.org/z/s7Th1f

Reply:
Please use `to_unsigned` instead of a cast.

Reply:
Handled in #2009